pkg/registry/apis/folders/validate.go

```diff
@@ -99,6 +99,11 @@ func validateOnUpdate(ctx context.Context,
 		return fmt.Errorf("k6 project may not be moved")
 	}
 
+	// moving to the root folder
+	if newParent == "" || newParent == folder.GeneralFolderUID {
```

Comment:
I'm not sure if we ever check permissions if you can move to root 🤔 

We do check it when the folder property exists:
https://github.com/grafana/grafana/blob/main/pkg/storage/unified/resource/server.go#L532

so maybe we need to explicitly check here?

or we could try something bigger where we always require the folder annotation set for resources that need annotation support:
https://github.com/grafana/grafana/compare/require-folder-annotation -- but this will also require updates to the auth client